projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-selection-handler.model.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  PopoverBackdrop,
+  PopoverFixedPositionLocation,
+  PopoverPositionType,
+  PopoverRef,
+  PopoverService
+} from '@hypertrace/components';
+import { BOOLEAN_PROPERTY, Model, ModelProperty } from '@hypertrace/hyperdash';
+import { ModelInject } from '@hypertrace/hyperdash-angular';
+import { Observable, of } from 'rxjs';
+import { CartesianSelectedData } from '../../../../../../public-api';
+import { InteractionHandler } from '../../../../interaction/interaction-handler';
+import { CartesianExplorerContextMenuComponent } from './cartesian-explorer-context-menu/cartesian-explorer-context-menu.component';
+
+import { CartesainExplorerNavigationService } from './cartesian-explorer-navigation.service';
+
+@Model({
+  type: 'cartesian-explorer-selection-handler'
+})
+export class CartesianExplorerSelectionHandlerModel<TData> implements InteractionHandler {
+  @ModelInject(CartesainExplorerNavigationService)
+  private readonly cartesainExplorerNavigationService!: CartesainExplorerNavigationService;
+
+  @ModelInject(PopoverService)
+  private readonly popoverService!: PopoverService;
+
+  public popover?: PopoverRef;
+
+  @ModelProperty({
+    key: 'show-context-menu',
+    displayName: 'Show Context Menu',
+    type: BOOLEAN_PROPERTY.type
+  })
+  public isContextMenuVisible: boolean = true;
```

Comment:
nit: Since this is set to true by default, we don't need to set it to true again in model jsons

Reply:
probably call it `showContextMenu`. `isContextMenuVisible` makes it sound like whether the context menu is currently visible or not